cmd/prometheus/main.go

```diff
@@ -555,7 +555,10 @@ func main() {
 				db, err := tsdb.Open(
 					cfg.localStoragePath,
 					log.With(logger, "component", "tsdb"),
-					prometheus.DefaultRegisterer,
+					prometheus.WrapRegistererWithPrefix(
```

Comment:
I am against this, applications should not add prefixes to the libraries they happen to use - that's what target labels are for.

Reply:
In the discussions that preceded this change (see https://github.com/prometheus/tsdb/pull/383 and in particular #4620), the understanding of the people in charge of this (@brancz @gouthamve @grobie) seems to be that the metrics provided by the TSDB library mean sufficiently different things if it is used for Prometheus vs. high frequency trading or unicorn counting so that they should have different prefixes. If the meaning changes, the metric name should change, not the target label. Thus, the important question is if the meaning actually changes or rather not, and I [asked it in that PR](https://github.com/prometheus/prometheus/pull/4620#issue-215945608) (last sentence).

I don't really have an own idea about the answer to this question, which I'd like to defer to @brancz @gouthamve @grobie @fabxc , who have all the history. Do you want to change the answer given in #4620 in light of the concern @brian-brazil has raised?

Reply:
I'd view it as the same as process_.

The case for adding a prefix is when you've got sharding (vertical or horizontal) and are likely also adding labels to distinguish the various different tsdb instances.

Reply:
To be honest I don't really care and my only comments were about requesting not to change the `prometheus_tsdb_` prefix currently present (which you never planned to change in the end).

I see arguments for both sides, but would ultimately agree with @brian-brazil and leave the prefix in place.

Reply:
I also think @brian-brazil's argument is fair.

Reply:
Thanks for all the feedback. I haven't heard from @gouthamve yet, but I guess if he felt strongly, he would have chimed in by now.
I'll close this PR and revert the change in prometheus/tsdb.